### DISCUSSION TITLE
[Item] Trouble with creation of citation to topItem

### DISCUSSION THREAD

**Discussion by JAC28** [windingwind/zotero-better-notes#1159] [Q&A] [answered]
Hello, 
maybe I am overlooking something important - but is there a simple way to add the citation of the topItem to a note? Unfortunately, I cannot understand the previous mostly Chinese discussions and check whether this topic was already discussed. 

 What I have found so far:

- `Zotero.URI.getItemURI(topItem)` returns the correct zotero  link, but not formatted
- There is `Zotero.BetterNotes.api.convert.item2citation()` exposed by the API, however, I am not able to find the right type of arguments to call this function. As seen below:

`${await Zotero.BetterNotes.api.convert.item2citation(topItem)}` gives *Template Preview Error: TypeError: citationIds.every is not a function*

**Comment by windingwind** [comment, marked as answer]
The `item2citation(citationIds, options)`'s first argument accepts any array of numbers, which is the id of the citation item.

To get the id of an item:

`topItem.id` (number)

**Reply by JAC28** [reply]
Works like a charm, thank you for the quick answer.
